### DISCUSSION TITLE
(Joomla 4) Disable unwanted scripts in Frontend

### DISCUSSION THREAD

**Discussion by coolcat-creations** [joomla/joomla-cms#32533] [General] [answered]
I think it should be possible to disable fontawesome and whatever in the frontend.
I have a clean Joomla 4 installation and it loads tons of unwanted and unneeded css and js.
(FontAwesome, Joomla alert, core.system.min.js, jquery,.... and and and)

How can this be changed in core to have a more performant initial setup?

**Comment by brianteeman** [comment]
It's all down to your template and the components etc being displayed. You can see an example at HTTPS://learnjoomla4.com of what can be done.

**Reply by coolcat-creations** [reply]
Thank you but your site also loads the scripts I mentioned. How can this be disabled? I also tried with chlodders lightning template but still all those css and js is loaded. Example: https://das-ist-nicht-mein-job.de/
(core.system.min, fontawesome, Joomla Alert)

**Reply by brianteeman** [reply]
It is not loading fontawesome or jquery. _(If you find those please let me know)_

In J4 its really a case of deciding what you want to load instead of removing things. For example

https://github.com/joomla/joomla-cms/blob/9ca4e5672c82839f0e7933452cd4ea0085f33a4c/templates/cassiopeia/joomla.asset.json#L9-L14

Here you can see that the template is saying that fontawesome is a dependency of the main template css.

If you dont want to use fontawesome then you simply dont tell your template to load it.

Other elements are loaded because you are using a component/module etc that requires them

Here you can see that by using the metismenu mod_menu override you are introducing a metismenu javascript file.

https://github.com/joomla/joomla-cms/blob/9ca4e5672c82839f0e7933452cd4ea0085f33a4c/templates/cassiopeia/html/mod_menu/dropdown-metismenu.php#L16-L18

BUT because its in the mod_menu override it is only loaded on a page that uses the module. If you dont want that script then you dont register it in the module.

Also you need to look at the size of the files that are loaded and not the quantity. Again using https://learnjoomla4.com as an example you will see how by simply selecting to include the files correctly you can achieve impressive performance.

![image](https://user-images.githubusercontent.com/1296369/109336659-a0cbcd00-785b-11eb-8db8-b0d8c967c927.png)

[gtmetrix report](https://gtmetrix.com/reports/learnjoomla4.com/zLDX9cy9/)

There are no places on this site that I am disabling any asset and there isnt even any cache enabled

**Reply by Chacapamac** [reply]
I know this is old @brianteeman, but I still don’t understand exactly what you need to do to unload “Awesome” 
I think I understand the role of the “joomla.asset.json” when place in the template or other component, but how you actually realize de override?

**Comment by Fedik** [comment, marked as answer]
All assets loaded via Joomla AssetManager, can be disabled via AssetManager https://docs.joomla.org/J4.x:Web_Assets

Example for fontawesome style: `$wa->disableStyle('fontawesome')`
For core script : `$wa->disableScript('core')`, however there a high chance that it will be re-enabled by one of core feature, example if  the document have a script options, translations, etc.

**Reply by coolcat-creations** [reply]
Thank you Fedik, I added now 

$wa  = $this->getWebAssetManager();
$wa->disableStyle('fontawesome');
$wa->disableScript('core');


to the lighting template and in cassiopeia but it's still loaded. Seems to be a bug then?


when i add
$wa->disableScript('joomla-alert');

I have the error:      0 There is no "joomla-alert" asset of a "script" type in the registry.

**Reply by Fedik** [reply]
I just showing a general way. Much depend from active template and extensions.

The "core" can be re-enabled somewhere else :wink: 
And "fontawesome" is dependency for `template.cassiopeia.ltr`(.rtl) so it also will be re-enabled for cassiopeia template .

> 0 There is no  "joomla-alert" asset of a "script" type in the registry

It a webcomponent, called `webcomponent.joomla-alert`, so `$wa->disableScript('webcomponent.joomla-alert');`
But this may be re-enabled from `joomla.system.message` layout.
Here you better make a layout override.

**Reply by coolcat-creations** [reply]
Thank you... It's quite not the user friendly approach I hoped to find but I try to create a template with just one style loaded and hope I will succeed somehow.

**Reply by migors** [reply]
@coolcat-creations Hi, have you managed disabling message.min.js and other files? 
Thanks

**Comment by drmenzelit** [comment]
The fontawesome in Lightning is being loaded by the template itself, the css comes from "Joomla" (media/vendor/.../fontawesome.css), because it is in core and the template doesn't need to deliver an own css for fontawesome.
It is possible to use Joomla without fontawesome, but only if you don't need frontend editing...

**Reply by brianteeman** [reply]
frontend editing should register fontawesome as an asset

**Reply by drmenzelit** [reply]
As far as I can see, it is not as asset at the moment... when you edit or create an article in the frontend and you use the editor-xtd buttons that open an iframe (e.g. CMS Content -> Article), the content doesn't look good if you haven't fontawesome loaded...

**Reply by brianteeman** [reply]
Sorry - lost in translation. I should have said

We should make frontend editing register fontawesome as an asset

**Reply by Fedik** [reply]
I just checked, it not only about a content editing, 
there many other places where the icons is used, example:

![Screenshot_2021-02-28_12-14-29](https://user-images.githubusercontent.com/1568198/109415005-d63cfb80-79be-11eb-9874-b3f944352da6.png)

> The fontawesome in Lightning is being loaded by the template itself

Technically we can do the same, however to disable it you still need to edit templates files. That  the same if you just edit cassiopeia/joomla.asset.json.
